src/v/crypto/ossl_context_service.cc

```diff
@@ -364,7 +364,8 @@ is_fips_mode ossl_context_service::fips_mode() const {
 }
 
 bool ossl_context_service::in_rp_fixture_test() const {
-    return ::getenv("RP_FIXTURE_ENV") != nullptr;
+    char* val = ::getenv("RP_FIXTURE_ENV");
+    return val != nullptr && *val != '\0';
```

Comment:
nit I would rather check for a known explicit value here (1 yes or anything really) instead of just empty vs non empty